Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs

```diff
@@ -31,9 +32,43 @@ public QuestionRepository(TrappistDbContext dbContext)
             return await _dbContext.Question.AnyAsync(x => x.Id == questionId);
         }
 
-        public async Task<ICollection<Question>> GetAllQuestionsAsync(string userId)
+        public async Task<ICollection<Question>> GetAllQuestionsAsync(string userId, int numberOfCalls, int categoryId, string difficultyLevel, string searchQuestion)
         {
-            return (await _dbContext.Question.Where(u => u.CreatedByUserId.Equals(userId)).Include(x => x.Category).Include(x => x.CodeSnippetQuestion).Include(x => x.SingleMultipleAnswerQuestion).ThenInclude(x => x.SingleMultipleAnswerQuestionOption).OrderByDescending(g => g.CreatedDateTime).ToListAsync());
+
+            if (categoryId == 0 && difficultyLevel.Equals("All"))
+            {
+                if (searchQuestion != null)
```

Comment:
please optimize the entire query fetching process. order by should be there when you use skip and take